src/expr.rs

```diff
@@ -682,11 +684,23 @@ fn rewrite_match(context: &RewriteContext,
             result.push_str(&snippet);
         }
     }
-
-    // We'll miss any comments etc. between the last arm and the end of the
-    // match expression, but meh.
+    let last_comment = context.snippet(mk_sp(arm_end_pos(&arms[arms.len() - 1]), span.hi));
+    let last_comment = match last_comment.find_uncommented(",") {
+        Some(n) => &last_comment[n+1..],
+        None => &last_comment[..],
+    };
+    let last_comment = match last_comment.find_uncommented("}") {
+        Some(n) => &last_comment[..n-1],
+        None => &last_comment[..],
+    };
+    let last_comment = last_comment.trim();
```

Comment:
I feel like this chunk of code is repeated at least once elsewhere in the codebase and ought to be factored out into a function if possible.